app/controllers/talks_controller.rb

```diff
@@ -8,12 +8,15 @@ def talk
   end
 
   def talks
-    @talks ||= search_against(Talk)
+    return @talks if defined? @talks
+
+    talks = search_against(Talk)
       .published
-      .tagged_with(params[:tag])
       .includes(:event, :speaker)
       .page(params[:page])
       .sorted_by_date
+
+    @talks = params[:tag] ? talks.tagged_with(params[:tag]) : talks
```

Comment:
maybe something like ?
`.public_send(params[:tag] ? :tagged_with, params[:tag] : to_a`

Reply:
What about method `untagged_talks`. And than

```ruby
@talks ||= params[:tag] ? untagged_talks.tagged_with(params[:tag]) : untagged_talks
```